pkg/pillar/cmd/volumemgr/handlevolume.go

```diff
@@ -137,6 +138,40 @@ func handleDeferredVolumeCreate(ctx *volumemgrContext, key string, config *types
 			}
 			created = true
 			status.FileLocation = zVolDevice
+			if !tgt.CheckTargetIBlock(status.Key()) {
+				log.Functionf("generating target and vhost for %s", status.Key())
+				serial := tgt.GenerateNaaSerial()
+				wwn := fmt.Sprintf("naa.%s", serial)
+				err := tgt.TargetCreateIBlock(zVolDevice, status.Key(), serial)
```

Comment:
Why can't you use createTargetVhost() here?

Reply:
Good point, modified